apps/charterafrica/src/payload/utils/mapLinkForToolsCollection.js

```diff
@@ -0,0 +1,17 @@
+import getPageUrlUsingPayload from "./getPageUrlUsingPayload";
+
+const mapLinkForToolsCollection = async ({ doc, req: { payload } }) => {
```

Comment:
We can have a single higher-order function that receives page slug and return a function that adds href.

I mean all these hooks look exactly the same with exception of page slug (`tools`, `contributors` and `organisations`), right

Reply:
Yes, and my question [here](https://github.com/CodeForAfrica/ui/pull/555#discussion_r1301832395) is why is this different?

Reply:
Yes, I know that code, and that's what I'm asking about. This is how we're loading a tool:

```js
  const { docs } = await api.getCollection(collection, {
    locale,
    where: {
      slug: {
        equals: slug,
      },
    },
  });
```
If found, `tool = docs[0]`, right? My questions is: since `tool` contain `contributors` i.e. `tools.contributors`, doesn't this call also trigger the `afterRead` hook on contributors?

Reply:
If it triggers the hook then we can just add a hook in Contributors collection the same way we’ve added on in Organisations, no?

Reply:
No you haven't @saminegash . Please review the current implementation.